### PR TITLE
Update_pipeline

### PR DESCRIPTION
**Pull request by dfalster**
Enhance build pipeline

* Add option for base R
* Add custom R code to build code

### CODE REVIEW

File: inst/support/build_furrr.whisker [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# Load data resources
+schema <- get_schema()
+resource_metadata <- get_schema("config/metadata.yml", "metadata")
+definitions <- get_schema("config/traits.yml", "traits")
+unit_conversions <- get_unit_conversions("config/unit_conversions.csv")
+taxon_list <- read_csv_char("config/taxon_list.csv")
+
+# Build sources
+dataset_ids <- {{dataset_ids_vector}}[1:20]
```
Comment (yangsophieee) [inline inst/support/build_furrr.whisker:17]: Why is this [1:20]?

File: inst/support/build_furrr.whisker [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  dataset_build(file_metadata, file_data,  
+                definitions = definitions,
+                unit_conversion_functions = unit_conversions,
+                schema = schema,
+                resource_metadata = resource_metadata,
+                taxon_list = taxon_list)
+}
+
+# sources <- purrr::map(dataset_ids, f)
```
Comment (yangsophieee) [inline inst/support/build_furrr.whisker:31]: Should these commented out lines from `purrr` be removed?

File: tests/testthat/test-setup.R [hunk 3]
```diff
@@ -630,6 +650,11 @@ test_that("`build_setup_pipeline` is working", {
   expect_length(austraits_raw$taxa, 14)
   expect_length(austraits$taxa, 14)
   expect_equal(nrow(austraits$taxa), nrow(austraits_raw$taxa))
+
+  # compare products from two methods, except build_info
+  v <- setdiff(names(austraits), "build_info")
+  expect_equal(tmp_env$austraits[v], austraits[v])
```
Comment (yangsophieee) [inline tests/testthat/test-setup.R:659]: Do we need a test for the `furrr` method?

File: R/process.R [hunk 4]
```diff
@@ -258,6 +260,49 @@ dataset_process <- function(filename_data_raw,
...
+dataset_build <- function(
+    filename_metadata,
+    filename_data_raw,
+    definitions,
+    unit_conversion_functions,
+    schema,
+    resource_metadata,
+    taxon_list,
+    filter_missing_values = TRUE) {
+  dataset_config <- dataset_configure(filename_metadata, definitions, unit_conversions)
```
Comment (yangsophieee) [inline R/process.R:304]: Should this argument be `unit_conversions_functions` not `unit_conversions`?

**Review by yangsophieee — APPROVED** [review]
Looks awesome! I cleaned up some formatting and warnings and notes from R CMD Check.

File: R/process.R [hunk 5]
```diff
@@ -70,7 +70,8 @@ dataset_configure <- function(
 #' \dontrun{
 #' dataset_process("data/Falster_2003/data.csv", dataset_configure("data/Falster_2003/metadata.yml",
```
Comment (yangsophieee) [inline R/process.R:71]: This example needs to be updated

File: R/setup.R [hunk 6]
```diff
@@ -1152,13 +1159,33 @@ build_setup_pipeline <- function(template = readLines(system.file("support", "re
...
     )
 
-  str <- whisker::whisker.render(template, vals)
-  writeLines(str, "remake.yml")
+  # Setup pipeline based on selected method for building
+  pipieline <- whisker::whisker.render(template, data = vals)
+
+  if (method %in% c("base", "furrr")) {
+    writeLines(pipieline, "build.R")
+    message(green("\t-> build using file `build.R`"))
```
Comment (yangsophieee) [inline R/setup.R:1176]: Would it be clearer to say "build *compilation* using file `build.R`"?

File: tests/testthat/test-setup.R (hunk 3 above)
Comment (dfalster) [reply tests/testthat/test-setup.R:657]: Ideally, but I've left it out for now. Can you first confirm if it works on your machine?

**Comment by dfalster** [comment]
Great suggestions and changes @yangsophieee . I've addressed these. Can you confirm if the furrr option works on your windows machine? I wasn't sure. Historically, getting the same system for parallel processing to work on windows and non-windows has been hard.

File: tests/testthat/test-setup.R (hunk 3 above)
Comment (yangsophieee) [reply tests/testthat/test-setup.R:657]: Yes it works

**Pull request merged**